nixos/modules/services/hardware/sane_extra_backends/brscan5.nix

```diff
@@ -0,0 +1,115 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.hardware.sane.brscan5;
+
+  netDeviceList = attrValues cfg.netDevices;
+
+  etcFiles = pkgs.callPackage ./brscan5_etc_files.nix { netDevices = netDeviceList; };
+
+  netDeviceOpts = { name, ... }: {
+
+    options = {
+
+      name = mkOption {
+        type = types.str;
+        description = ''
+          The friendly name you give to the network device. If undefined,
+          the name of attribute will be used.
+        '';
+
+        example = literalExample "office1";
+      };
+
+      model = mkOption {
+        type = types.str;
+        description = ''
+          The model of the network device.
+        '';
+
+        example = literalExample "ADS-1200";
+      };
+
+      ip = mkOption {
+        type = with types; nullOr str;
+        default = null;
+        description = ''
+          The ip address of the device. If undefined, you will have to
+          provide a nodename.
+        '';
+
+        example = literalExample "192.168.1.2";
+      };
+
+      nodename = mkOption {
+        type = with types; nullOr str;
+        default = null;
+        description = ''
+          The node name of the device. If undefined, you will have to
+          provide an ip.
+        '';
+
+        example = literalExample "BRW0080927AFBCE";
+      };
+
+    };
+
+
+    config =
+      { name = mkDefault name;
+      };
+  };
+
+in
+
+{
+  options = {
+
+    hardware.sane.brscan5.enable =
+      mkEnableOption "Brother's brscan5 scan backend" // {
```

Comment:
Does this actually render the way one might think it would? :thinking:

Reply:
`mkEnableOption "thing" // {description = ...` is done a few other modules (notably, the other sane_extra_backends):
```
nixos/modules/programs/java.nix
nixos/modules/programs/x2goserver.nix
nixos/modules/services/hardware/sane_extra_backends/brscan4.nix
nixos/modules/services/hardware/sane_extra_backends/dsseries.nix
nixos/modules/services/misc/gitea.nix
nixos/modules/services/networking/i2pd.nix
nixos/modules/services/networking/wpa_supplicant.nix
nixos/modules/services/web-apps/wiki-js.nix
nixos/modules/services/x11/xautolock.nix
nixos/modules/virtualisation/virtualbox-host.nix
```

Since this seems unusual I added a commit to simplify this option.

Thanks!